File: Programmation/TD4/ex3.pl
#!/usr/bin/perl
use warnings;
use strict;

my (@noms, $nom, @pluriels, $pluriel) ;
my ($index, $noms) ;

open(INVAR, '<', "noms-invariables.txt")
    or die "Vous devez avoir un fichier contenant les noms invariables !\n" ;
my @invar = <INVAR> ; # is needed cause you can't grep files (or else use File::Grep)

print "Entrez autant de noms commun masculin singulier que vous le voulez
(Tappez \"X\" pour continuer) : \n" ;

while (1)
{
  chomp($nom = <STDIN>) ;
  last if ($nom eq "X") ;
  @noms = (@noms, $nom) ;
}


# définition des pluriels
while (($index, $nom) = each @noms){
    ($pluriel = $nom) ;
    ($pluriel = $nom . "s") if (!grep (/$nom\n/, @invar ));
    @pluriels = (@pluriels, $pluriel);

    print "\nLes flexions de \"" , $nom , "\" sont :\n" ;
    print "un ", $nom, ", des " , $pluriel , "\n" ;
}

File: Programmation/TD4/ex3.rb
#!/usr/bin/env ruby

text = File.new("noms-invariables.txt").read

puts "Type mascular french words,
comma separated :"

# getting string, splitting and converting to hash
flexions = Hash[gets.chomp.split(/, |,/).map {|val| [val, val]}]

flexions.each_key do |word|
  unless text =~ /\b#{word}\b/
      flexions[word] = word + "s"
  end
end

flexions.each do |word, plural|
  puts "\n"
  puts "Flexions of #{word} are :"
  puts "Un #{word}, des #{plural}"
end
